bootstrap.sh

```diff
@@ -2,6 +2,12 @@
 
 # GROS MÉNAGE
 cd ~
+
+# Backup existing dotfiles
+mkdir .jumpbox_dotfile_backups
+cp {.bash*,.bin*,.cache*,.chef*,.config*,.gitconfig,.lesshst,.oh-my-zsh,.profile,.ssh/known_hosts,.vim*,.zcomp*,.zsh*} .jumpbox_dotfile_backups/
```

Comment:
this would fail if any of these files do not exist, can you add a test to make sure the files exists before copying them?

Reply:
Good point!
I've updated it to check for files and directories and copy them appropriately.

Example output:

```
Creating backups of your dotfiles at ~/.jumpbox_dotfile_backups.
Running this again will replace these backups with the files this generates.

~/.config copied to ~/.jumpbox_dotfile_backups/.config
~/.oh-my-zsh copied to ~/.jumpbox_dotfile_backups/.oh-my-zsh
~/.ssh/known_hosts copied to ~/.jumpbox_dotfile_backups/.ssh/known_hosts
~/.viminfo copied to ~/.jumpbox_dotfile_backups/.viminfo
~/.vimrc copied to ~/.jumpbox_dotfile_backups/.vimrc
~/.zcompdump-COMP11583-5.3 copied to ~/.jumpbox_dotfile_backups/.zcompdump-COMP11583-5.3
~/.zcompdump-COMP11583-5.7.1 copied to ~/.jumpbox_dotfile_backups/.zcompdump-COMP11583-5.7.1
~/.zsh_history copied to ~/.jumpbox_dotfile_backups/.zsh_history
~/.zshenv copied to ~/.jumpbox_dotfile_backups/.zshenv
~/.zshrc copied to ~/.jumpbox_dotfile_backups/.zshrc
~/.zshrc_cq_1 copied to ~/.jumpbox_dotfile_backups/.zshrc_cq_1
Removing/replacing dotfiles which have been backed up.
```

One thing I've caught- it doesnt handle symlinks (Theres a .bin left around in my ~/ from last time I ran this), though I think that's edge-casey enough to not worry about.